packages/by-name/contrast-node-installer/package.nix

```diff
@@ -16,21 +16,27 @@ buildGoModule {
   src =
     let
       inherit (lib) fileset path hasSuffix;
-      root = ../../../node-installer;
+      root = ../../../.;
+      node-installer = ../../../node-installer;
     in
     fileset.toSource {
       inherit root;
       fileset = fileset.unions [
         (path.append root "go.mod")
         (path.append root "go.sum")
-        (fileset.fileFilter (file: hasSuffix ".toml" file.name) root)
-        (fileset.fileFilter (file: hasSuffix ".toml.tmpl" file.name) root)
-        (fileset.fileFilter (file: hasSuffix ".go" file.name) root)
+        (fileset.fileFilter (file: hasSuffix ".toml" file.name) node-installer)
+        (fileset.fileFilter (file: hasSuffix ".toml.tmpl" file.name) node-installer)
+        (fileset.fileFilter (file: hasSuffix ".go" file.name) node-installer)
+        # To prevent unnecessary rebuilds, and since the platforms package is the only
+        # imported package from the Contrast main module by the node-installer, we explicitly
+        # supply it here, rather than including "internal" as a whole.
+        (path.append root "internal/platforms")
```

Comment:
I don't think this is a good idea, it will break the next person importing another package in a very cursed way.

Reply:
It's a decision between having a lot of rebuilds and being developer-friendly.

I think having less rebuilds is better here, since the node-installer is less likely to be changed (especially with further additions from the Contrast package) than the other way around. What about adding a comment that explains precisely what to do to add another package instead?